eng/validation/update-channel.ps1

```diff
@@ -15,34 +15,45 @@ function Get-Headers([string]$accept, [string]$barToken) {
-$getAssetApiEndpoint = "$maestroEndpoint/api/assets?name=$arcadeSdkPackageName&version=$arcadeSdkVersion&api-version=2018-07-16"
+$getAssetsApiEndpoint = "$maestroEndpoint/api/assets?name=$arcadeSdkPackageName&version=$arcadeSdkVersion&api-version=2018-07-16"
 $headers = Get-Headers 'text/plain' $barToken
 
 # Get the Microsoft.DotNet.Arcade.Sdk with the version $arcadeSdkVersion so we can get the id of the build
-$assets = Invoke-WebRequest -Uri $getAssetApiEndpoint -Headers $headers | ConvertFrom-Json
-
-if ($assets) {
-    # Today, there shouldn't be more that one build linked to a given Arcade SDK version, but if in the future we only bump the version if
-    # changes were done that could be a possibility. Also, since we already validated version $arcadeSdkVersion we move all the builds that
-    # produced it to channel $channelId
-    foreach ($asset in $assets) {
-        try {
-            $buildId = $asset[0].'buildId'
-
-            $postBuildIntoChannelApiEndpoint = "$maestroEndpoint/api/channels/$channelId/builds/$buildId/?name=$arcadeSdkPackageName&version=$arcadeSdkVersion&api-version=2018-07-16"
-            $headers = Get-Headers 'application/json' $barToken
-        
-            Write-Host "POSTing to $postBuildIntoChannelApiEndpoint..."
-            $postResponse = Invoke-WebRequest -Uri $postBuildIntoChannelApiEndpoint -Headers $headers -Method Post
-            Write-Host "Build '$buildId' was successfully added to channel '$channelId'"
-         }
-        catch {
-            Write-Host $_
-            Write-Host $_.ScriptStackTrace
-            exit 1
-        }
-    }
-} else {
+$assets = Invoke-WebRequest -Uri $getAssetsApiEndpoint -Headers $headers | ConvertFrom-Json
+
+if (!$assets) {
     Write-Host "Asset '$arcadeSdkPackageName' with version $arcadeSdkVersion was not found"
     exit 1
-}
\ No newline at end of file
+}
+
+if ($assets.Count > 1) {
+    Write-Host "More than 1 asset matched the version '$arcadeSdkVersion' of Microsoft.DotNet.Arcade.Sdk. This is not normal. Stopping execution..."
+    exit 1
+}
+
+$getChannelsEndpoint = "$maestroEndpoint/api/channels?api-version=2018-07-16"
+$channels = Invoke-WebRequest -Uri $getChannelsEndpoint -Headers $headers | ConvertFrom-Json
+$matchedChannel = $($channels | Where-Object -Property "name" -Value "${targetChannelName}" -EQ | Select-Object -Property id)
+
+if (!$matchedChannel) {
+    Write-Host "Channel with name '$targetChannelName' was not found..."
+    exit 1    
+}
+
+$channelId = $matchedChannel.id
+
+try {
+    $buildId = $assets[0].'buildId'
+
+    $postBuildIntoChannelApiEndpoint = "$maestroEndpoint/api/channels/$channelId/builds/$buildId/?name=$arcadeSdkPackageName&version=$arcadeSdkVersion&api-version=2018-07-16"
+    $headers = Get-Headers 'application/json' $barToken
+        
+    Write-Host "POSTing to $postBuildIntoChannelApiEndpoint..."
+    $postResponse = Invoke-WebRequest -Uri $postBuildIntoChannelApiEndpoint -Headers $headers -Method Post
```

Comment:
As discussed offline. One further improvement is to check if the build is already in the Latest channel. If it is, we shouldn't try to post it, and it shouldn't be considered an error.

Reply:
https://dnceng.visualstudio.com/internal/_build/results?buildId=61149 implements this

Reply:
Is that code in this branch? I don't see those messages in the PR.